pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3663,8 +3664,18 @@ func (c *Cloud) GetLoadBalancer(ctx context.Context, clusterName string, service
 
 // GetLoadBalancerName is an implementation of LoadBalancer.GetLoadBalancerName
 func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
-	return cloudprovider.DefaultLoadBalancerName(service)
+	// Ref: https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CreateLoadBalancer.html
+	// unique for the region / max 32 chars / alphanumeric plus hyphen / can't start or end with hyphen
+	reg := regexp.MustCompile(`[^a-zA-Z0-9\-]+`)
+
+	ret := fmt.Sprintf("%s-%s", clusterName, string(service.UID))
+	ret = reg.ReplaceAllString(ret, "")
+
+	if len(ret) > 32 {
+		ret = ret[:32]
```

Comment:
Any cluster with a name 31 characters long will result in the ELB name for all LoadBalancers in the cluster being the same. I'm pretty sure that's not what you want, right?

Reply:
Yup, unfortunately, that's the char limit for LB names. :(